modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web-test/src/testIntegration/java/com/liferay/portal/search/tuning/rankings/web/internal/upgrade/BaseRankingUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,155 @@
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.search.tuning.rankings.web.internal.upgrade;
+
+import com.liferay.change.tracking.test.util.BaseCTUpgradeProcessTestCase;
+import com.liferay.counter.kernel.service.CounterLocalService;
+import com.liferay.json.storage.model.JSONStorageEntry;
+import com.liferay.json.storage.service.JSONStorageEntryLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.change.tracking.CTModel;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.change.tracking.CTService;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.search.tuning.rankings.index.Ranking;
+import com.liferay.portal.search.tuning.rankings.index.name.RankingIndexName;
+import com.liferay.portal.search.tuning.rankings.index.name.RankingIndexNameBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+
+/**
+ * @author Almir Ferreira
+ */
+public abstract class BaseRankingUpgradeProcessTestCase
+	extends BaseCTUpgradeProcessTestCase {
+
+	@BeforeClass
+	public static void setUp() throws PortalException {
+		_companyId = TestPropsValues.getCompanyId();
+		rankingClassNameId = classNameLocalService.getClassNameId(
+			Ranking.class);
+		_rankingIndexName = _rankingIndexNameBuilder.getRankingIndexName(
+			_companyId);
+	}
+
+	@After
+	public void tearDown() {
+		Iterator<Long> iterator = _jsonStorageEntryClassPKs.iterator();
+
+		while (iterator.hasNext()) {
+			jsonStorageEntryLocalService.deleteJSONStorageEntries(
+				rankingClassNameId, iterator.next());
+
+			iterator.remove();
+		}
+	}
+
+	@Override
+	protected CTModel<?> addCTModel() {
+		JSONStorageEntry jsonStorageEntry =
+			jsonStorageEntryLocalService.createJSONStorageEntry(
+				counterLocalService.increment());
+
+		jsonStorageEntry.setClassNameId(rankingClassNameId);
+		jsonStorageEntry.setClassPK(1);
+		jsonStorageEntry.setKey("inactive");
+		jsonStorageEntry.setValue(true);
+
+		return jsonStorageEntryLocalService.updateJSONStorageEntry(
+			jsonStorageEntry);
+	}
+
+	protected void addRanking(long classNameId, long classPK) {
+		String rankingDocumentId =
+			Ranking.class.getName() + "_PORTLET_" + classPK;
+
+		jsonStorageEntryLocalService.addJSONStorageEntries(
+			_companyId, classNameId, classPK,
+			JSONUtil.put(
+				"aliases", ""
+			).put(
+				"groupExternalReferenceCode", ""
+			).put(
+				"hiddenDocumentIds", ""
+			).put(
```

Comment:
I noticed that inactive is set here. Since it will be removed in the upgrade, using it might not accurately reflect a "real ranking" in future upgrades.  Can we explore alternative approaches that avoid using inactive if it's not strictly necessary?

Reply:
We talked about this and came to the following conclusion: 
We have alternatives such as passing JSON as a parameter in the "addRanking" function or using resource files containing the JSON to be used in the test. Considering that both tests use the same JSON, both alternatives become unfeasible as they would result in unnecessary code duplication. We believe that implementing one of these alternatives has to be done when a new upgrade is added.